torchinductor/codegen/triton_conv_delta_x_hwc.j2

```diff
@@ -0,0 +1,219 @@
+        delta_xh = tl.load(delta_xh_ptrs, mask=off_x_crs < CRS)
+        delta_xw = tl.load(delta_xw_ptrs, mask=off_x_crs < CRS)
+        delta_xc = tl.load(delta_xc_ptrs, mask=off_x_crs < CRS)
+        off_x_crs_unpacked = (
+            delta_xh * stride_xh + delta_xw * stride_xw + delta_xc * stride_xc
+        )
+        x_ptrs = x + off_x_nhw[:, None] + off_x_crs_unpacked[None, :]
+    else:
+        x_ptrs = x + off_x_nhw[:, None] + off_x_crs[None, :]
+        delta_xh = 0
+        delta_xw = 0
+
+    mask_x = (
+        (off_x_n < BATCH)[:, None]
+        & (off_x_crs < CRS)[None, :]
+        & (off_x_h[:, None] + delta_xh[None, :] >= 0)
+        & (off_x_h[:, None] + delta_xh[None, :] < IN_H)
+        & (off_x_w[:, None] + delta_xw[None, :] >= 0)
+        & (off_x_w[:, None] + delta_xw[None, :] < IN_W)
+    )
+
+    # offset for the inital ptr for w
+    off_w_crs = tl.arange(0, BLOCK_K)
+    off_w_k = off_y_k
+    w_ptrs = w + off_w_crs[:, None] + off_w_k[None, :] * stride_wn
+    mask_w = (off_x_crs < CRS)[:, None] & (off_w_k < KERNEL_N)[None, :]
+
+    # ------ load x ------
+    matrix_x = tl.load(x_ptrs, mask=mask_x)
+    # ------ load w ------
+    matrix_w = tl.load(w_ptrs, mask=mask_w)
+
+    # -----------------------------------------------------------
+    # allocate accumulator
+    acc = tl.zeros((BLOCK_M, BLOCK_N), dtype=ACC_TYPE)
+    for crs in range(0, CRS, BLOCK_K):
+
+        # ------ matrix multiplication ------
+        acc += tl.dot(matrix_x, matrix_w)
+        # ------ update ptrs ------
+        w_ptrs += BLOCK_K
+        # load inc ptr of x, upade x_ptrs
+        off_x_crs = crs + BLOCK_K + tl.arange(0, BLOCK_K)
+        if not CONV1X1_NHWC:
+            delta_xh_ptrs += BLOCK_K
+            delta_xw_ptrs += BLOCK_K
+            delta_xc_ptrs += BLOCK_K
+            delta_xh = tl.load(delta_xh_ptrs, mask=off_x_crs < CRS)
+            delta_xw = tl.load(delta_xw_ptrs, mask=off_x_crs < CRS)
+            delta_xc = tl.load(delta_xc_ptrs, mask=off_x_crs < CRS)
+            off_x_crs_unpacked = (
+                delta_xh * stride_xh + delta_xw * stride_xw + delta_xc * stride_xc
+            )
+            x_ptrs = x + off_x_nhw[:, None] + off_x_crs_unpacked[None, :]
+        else:
+            x_ptrs += BLOCK_K
+
+        mask_x = (
+            (off_x_n < BATCH)[:, None]
+            & (off_x_crs < CRS)[None, :]
+            & (off_x_h[:, None] + delta_xh[None, :] >= 0)
+            & (off_x_h[:, None] + delta_xh[None, :] < IN_H)
+            & (off_x_w[:, None] + delta_xw[None, :] >= 0)
+            & (off_x_w[:, None] + delta_xw[None, :] < IN_W)
+        )
+        mask_w = (off_x_crs < CRS)[:, None] & (off_w_k < KERNEL_N)[None, :]
+        # ------ prefetch ------
+        # ------ load x ------
+        matrix_x = tl.load(x_ptrs, mask=mask_x)
+        # ------ load w ------
+        matrix_w = tl.load(w_ptrs, mask=mask_w)
+
+    # add bias if is not None
+    if bias is not None:
```

Comment:
We should be able to drop bias support (we split it out into a separate kernel in lowering)

Reply:
Removed bias